addons/mail/static/src/js/client_action.js

```diff
@@ -287,6 +356,41 @@ var ChatAction = Widget.extend(ControlPanelMixin, {
                 self.thread.scroll_to({offset: offset});
             });
     },
+    /**
+     * @private
+     */
+    _onModeration: function (messageID, decision) {
+        if (decision === 'reject') {
+            this._onRejectMessage([messageID]);
+        } else if (decision === 'discard') {
+            this._onDiscardMessage([messageID]);
```

Comment:
never directly calls something that looks like a handler. onDiscardMessage should be a private method